covalent_slurm_plugin/slurm.py

```diff
@@ -201,28 +254,11 @@ async def get_status(self, info_dict: dict) -> Union[Result, str]:
         job_id = info_dict.get("job_id")
         if job_id is None:
             return Result.NEW_OBJ
+        
+        proc = await conn.run(f"export PATH=$PATH:/opt/slurm/bin && scontrol show job {job_id}")
```

Comment:
this line assumes that slurm is installed under `/opt/slurm/bin`. Might be better to use `shutil.which('slurm')` to find the actual path on the remote machine. This will cover the cases for machines where slurm is not installed under `/opt/`

Reply:
So, the issue is that it couldn't find slurm commands since they are not in the `PATH`, which will fail even with `shutil.which('slurm')`. Is there a specific time when slurm commands are loaded in the path?

Reply:
in some cases, like in our setup, the path is provided in `/etc/profile.d` so it's sufficient to do `source /etc/profile`, although in many real clusters users will need to load the slurm module manually (usually in a bashrc file).  Maybe let's try both here:
```
source /etc/profile
module whatis slurm &> /dev/null
if [ $? -eq 0 ] ; then
  module load slurm
fi 
```
A more robust solution would be to provide an optional parameter `slurm_path` that users can specify if either of these other methods do not work.

Reply:
the `module` cmd must also be in the path... but if on systems environment modules are installed properly, sourcing /etc/profile would work... 

@wjcunningham7 agreed a more robust implementation would be to require a `slurm_path` parameter from the user end, that way we don't have to worry about SLURM being installed on different system locations

Reply:
I see, it seems like the best way forward is to assume slurm commands are already sourced in bashrc, if not then users can provide `slurm_path` which will be added to `PATH` before executing the slurm commands. Is that a reasonable solution?

Reply:
@kessler-frost that sounds good to me... will let @wjcunningham7 decide further course of action... otherwise PR LGTM!

Reply:
Got it, that means we (at Agnostiq) will have to specify `slurm_path` whenever we use this executor. Unless the user (one of us) has already added `source /etc/profile` to their `.bashrc`.

Reply:
Marking this as resolved.

Reply:
> we should use the code snippet I provided above, but if which sbatch fails afterwards we should return a message that the user needs to provide slurm_path to the executor's constructor. If the slurm_path is provided, skip that code snippet and directly validate which sbatch

Hmm, I see, I'll add that. So the assumption is `module` is a recognized command by the remote shell.

> .bashrc is entirely unrelated here, and will not be sourced at any point before the task is run

I tried adding `source /etc/profile` to `/federation/sankalp/.bashrc` and `conn.run("which sbatch")` gives the correct sbatch path which it wasn't giving before.

Reply:
Got it, thanks for the explanation.

Reply:
Implemented suggested changes, resolving this now.